x-pack/plugins/apm/public/components/app/correlations/correlations_chart.tsx

```diff
@@ -0,0 +1,202 @@
+
+import { useTheme } from '../../../hooks/use_theme';
+import { HistogramItem } from '../../../../common/search_strategies/correlations/types';
+
+const { euiColorMediumShade } = euiVars;
+const axisColor = euiColorMediumShade;
+
+const axes: RecursivePartial<AxisStyle> = {
+  axisLine: {
+    stroke: axisColor,
+  },
+  tickLabel: {
+    fontSize: 10,
+    fill: axisColor,
+    padding: 0,
+  },
+  tickLine: {
+    stroke: axisColor,
+    size: 5,
+  },
+  gridLine: {
+    horizontal: {
+      dash: [1, 2],
+    },
+    vertical: {
+      strokeWidth: 1,
+    },
+  },
+};
+const chartTheme: PartialTheme = {
+  axes,
+  legend: {
+    spacingBuffer: 100,
+  },
+  areaSeriesStyle: {
+    line: {
+      visible: false,
+    },
+  },
+};
+
+// Default axis to start at 0 for latency correlations chart
+const xAxisDomain = {
+  min: 0,
+};
+
+interface CorrelationsChartProps {
+  field?: string;
+  value?: string;
+  histogram?: HistogramItem[];
+  markerValue: number;
+  markerPercentile: number;
+  overallHistogram: HistogramItem[];
+}
+
+const annotationsStyle = {
+  line: {
+    strokeWidth: 1,
+    stroke: 'gray',
+    opacity: 0.8,
+  },
+  details: {
+    fontSize: 8,
+    fontFamily: 'Arial',
+    fontStyle: 'normal',
+    fill: 'gray',
+    padding: 0,
+  },
+};
+
+export function CorrelationsChart({
+  field,
+  value,
+  histogram,
+  markerValue,
+  markerPercentile,
+  overallHistogram,
+}: CorrelationsChartProps) {
+  const euiTheme = useTheme();
+
+  if (!Array.isArray(overallHistogram)) return <div />;
+  const annotationsDataValues: LineAnnotationDatum[] = [
+    { dataValue: markerValue, details: `${markerPercentile}th percentile` },
+  ];
+
+  const xMax = max(overallHistogram.map((d) => d.key)) ?? 0;
+  const durationFormatter = getDurationFormatter(xMax);
+
+  return (
+    <div style={{ overflow: 'hidden', textOverflow: 'ellipsis' }}>
+      <Chart
+        size={{
+          height: '250px',
+        }}
+      >
+        <Settings
+          rotation={0}
+          theme={chartTheme}
+          showLegend
+          legendPosition={Position.Bottom}
+          xDomain={xAxisDomain}
+        />
+
+        <LineAnnotation
+          id="annotation_1"
+          domainType={AnnotationDomainType.XDomain}
+          dataValues={annotationsDataValues}
+          style={annotationsStyle}
+          marker={`${markerPercentile}p`}
+          markerPosition={'top'}
+        />
+
+        <Axis
+          id="x-axis"
+          title=""
+          position={Position.Bottom}
+          tickFormat={(d) => durationFormatter(d).formatted}
+        />
+        <Axis
+          id="y-axis"
+          title={i18n.translate(
+            'xpack.apm.correlations.latency.chart.numberOfTransactionsLabel',
+            { defaultMessage: '# transactions' }
+          )}
+          position={Position.Left}
+          tickFormat={(d) => (d === 0 || d === 10 || d % 100 === 0 ? d : '')}
```

Comment:
Can't you simply specify the ticks?

```suggestion
          ticks={[0, 10, 100]}

```

Reply:
It should be everything divisible by 100 as well for the tick labeling

Reply:
I ended up the logic to only show tick labels in powers of 10 (e.g. 0, 10, 100, 1000, ... and so on) here https://github.com/elastic/kibana/pull/99905/commits/9b73f97d733b265c9a4b9e04849b2c897d1aaaf8